bindings/py/cpp_src/bindings/sdr/py_SDR.cpp

```diff
@@ -322,9 +322,12 @@ Argument dimensions A list of dimension sizes, defining the shape of the SDR.)",
             { return new Reshape(self, dimensions); },
 R"(See class nupic.bindings.sdr.Reshape)");
 
+        py_SDR.def("flatten", [](SDR &self)
+            { return new Reshape(self, {self.size}); },
```

Comment:
will we use this? But it's a common function such as in np

Reply:
This is useful because every function checks the dimensions of its input SDRs, but most of the time ppl just care about the size and ignore the dimensions.

For example:
```Python
A = SDR( [1,2,3,4,5,6,7] ) # awful dimensions
B = SDR( [77] ) # Incompatible dimensions
C = SDR( A.size + B.size ).concatenate(
                             A.flatten(), B.flatten() )
```

Reply:
Also some algorithms require annoying dimensions, for example the Temporal Memory appends a dimension for the cells in the mini-columns.